src/Testing.NSubstitute/AutoSubstituteTest.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using DryIoc;
+using DryIoc.Microsoft.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Serilog;
+using Serilog.Events;
+using Serilog.Extensions.Logging;
+using Xunit.Abstractions;
+
+namespace Rocket.Surgery.Extensions.Testing
+{
+    public abstract class AutoSubstituteTest : LoggerTest
+    {
+        protected AutoSubstituteTest(ITestOutputHelper outputHelper, string logFormat = "[{Timestamp:HH:mm:ss} {Level:w4}] {Message}{NewLine}{Exception}", Action<LoggerConfiguration>? configureLogger = null)
+            : this(outputHelper, LogLevel.Information, logFormat, configureLogger) { }
+
+        protected AutoSubstituteTest(ITestOutputHelper outputHelper, LogLevel minLevel, string logFormat = "[{Timestamp:HH:mm:ss} {Level:w4}] {Message}{NewLine}{Exception}", Action<LoggerConfiguration>? configureLogger = null)
+            : this(outputHelper, LevelConvert.ToSerilogLevel(minLevel), logFormat, configureLogger) { }
+
+        protected AutoSubstituteTest(ITestOutputHelper outputHelper, LogEventLevel minLevel, string logFormat = "[{Timestamp:HH:mm:ss} {Level:w4}] {Message}{NewLine}{Exception}", Action<LoggerConfiguration>? configureLogger = null)
+            : base(outputHelper, minLevel, logFormat, configureLogger) => AutoSubstitute = new AutoSubstitute(configureAction: ConfigureContainer);
+
+        /// <summary>
+        /// The Configuration if defined otherwise empty.
+        /// </summary>
+        protected IConfiguration Configuration { get; private set; } = new ConfigurationBuilder().Build();
+
+        /// <summary>
+        /// The AutoFake instance
+        /// </summary>
+        protected AutoSubstitute AutoSubstitute { get; }
+
+        /// <summary>
+        /// The AutoFake instance
+        /// </summary>
+        protected AutoSubstitute NSubstitute => AutoSubstitute;
```

Comment:
So I removed this from AutoFake because it got in the way or namespaces and globals and stuff, so you probably don't want this here.  It was a terrible decision on my part, basically because intellisense always wanted to auto complete `autof` to `Autofac` vs `AutoFake` but that is no longer a concern since we're using DryIoc now.

Reply:
I noticed this, and I had a similar naming issue.  Glad you pointed out you want to move away from this pattern.